test/correctness/inlined_generator.cpp

```diff
@@ -0,0 +1,86 @@
+#include "Halide.h"
+
+using namespace Halide;
+
+void verify(const Buffer<int32_t> &img, float compiletime_factor, float runtime_factor, int runtime_offset) {
+    img.for_each_element([=](int x, int y, int c) {
+        int expected = (int32_t)(compiletime_factor * runtime_factor * c * (x > y ? x : y)) + runtime_offset;
+        int actual = img(x, y, c);
+        assert(expected == actual);
+    });
+}
+
+class Example : public Generator<Example> {
+public:
+    using Generator<Example>::Generator;
+
+    enum class SomeEnum { Foo, Bar };
+
+    GeneratorParam<float> compiletime_factor{ "compiletime_factor", 1, 0, 100 };
+
+    ScheduleParam<bool> vectorize{ "vectorize", true };
+
+    Input<float> runtime_factor{ "runtime_factor", 1.0 };
+    Input<int> runtime_offset{ "runtime_offset", 0 };
+
+    Output<Func> output{ "output", Int(32), 3 };
+
+    void generate() {
+        Func f;
+        f(x, y) = max(x, y);
+        output(x, y, c) = cast(output.type(), f(x, y) * c * compiletime_factor * runtime_factor + runtime_offset);
+    }
+
+    void schedule() {
+        output.bound(c, 0, 3).reorder(c, x, y).unroll(c);
+        output.specialize(vectorize).vectorize(x, natural_vector_size(output.type()));
+    }
+
+private:
+    Var x{"x"}, y{"y"}, c{"c"};
+};
+
+int main(int argc, char **argv) {
+    JITGeneratorContext context(get_target_from_environment());
```

Comment:
> Should probably be get_jit_target_from_environment.

Yep.

> Given that it's a jit context, should there be a zero-arg constructor that just uses that? 

I thought about that, but I'm not entirely convinced, mainly because I think that any code that ever gets config from envvars should be as explicit about it as possible, to avoid creeping implicit behaviors that require specific build setups. (I realize that existing Halide APIs don't have this attitude, and thus we don't see eye-to-eye on this principle....)